gapis/api/vulkan/find_issues.go

```diff
@@ -55,17 +88,193 @@ func (t *findIssues) reportTo(r replay.Result) { t.res = append(t.res, r) }
+		layers := []Charᶜᵖ{}
+		for _, l := range info.PpEnabledLayerNames().Slice(0, uint64(layerCount), l).MustRead(ctx, cmd, s, nil) {
+			if !isValidationLayer(l.String()) {
+				layers = append(layers, l)
+			}
+		}
+
+		validationLayersData := []api.AllocResult{}
+		for _, v := range validationLayers {
+			d := s.AllocDataOrPanic(ctx, v)
+			validationLayersData = append(validationLayersData, d)
+			layers = append(layers, NewCharᶜᵖ(d.Ptr()))
+		}
+		layersData := s.AllocDataOrPanic(ctx, layers)
+
+		extCount := info.EnabledExtensionCount()
+		exts := info.PpEnabledExtensionNames().Slice(0, uint64(extCount), l).MustRead(ctx, cmd, s, nil)
+		var debugReportExtNameData api.AllocResult
+		hasDebugReport := false
+		for _, e := range exts {
+			if e.String() == debugReportExtension {
+				hasDebugReport = true
+			}
+		}
+		if !hasDebugReport {
+			debugReportExtNameData = s.AllocDataOrPanic(ctx, debugReportExtension)
+			exts = append(exts, NewCharᶜᵖ(debugReportExtNameData.Ptr()))
+		}
+		extsData := s.AllocDataOrPanic(ctx, exts)
+
+		info.SetEnabledLayerCount(uint32(len(layers)))
+		info.SetPpEnabledLayerNames(NewCharᶜᵖᶜᵖ(layersData.Ptr()))
+		info.SetEnabledExtensionCount(uint32(len(exts)))
+		info.SetPpEnabledExtensionNames(NewCharᶜᵖᶜᵖ(extsData.Ptr()))
+		infoData := s.AllocDataOrPanic(ctx, info)
+
+		newCmd := cb.VkCreateInstance(infoData.Ptr(), cmd.PAllocator(), cmd.PInstance(), cmd.Result())
+		for _, d := range validationLayersData {
+			newCmd.AddRead(d.Data())
+		}
+		newCmd.AddRead(debugReportExtNameData.Data())
+		newCmd.AddRead(infoData.Data())
+		newCmd.AddRead(layersData.Data())
+		newCmd.AddRead(extsData.Data())
+		// Also add back all the other read/write observations of the origianl vkCreateInstance
+		for _, r := range cmd.Extras().Observations().Reads {
+			newCmd.AddRead(r.Range, r.ID)
+		}
+		for _, w := range cmd.Extras().Observations().Writes {
+			newCmd.AddWrite(w.Range, w.ID)
+		}
+		out.MutateAndWrite(ctx, id, newCmd)
+
+	// Modify the vkCreateDevice to remove any validation layers and add individual
+	// layers back later. Same reason as vkCreateInstance
+	case *VkCreateDevice:
+		cmd.Extras().Observations().ApplyReads(s.Memory.ApplicationPool())
+		info := cmd.PCreateInfo().MustRead(ctx, cmd, s, nil)
+		layerCount := info.EnabledLayerCount()
+		layers := []Charᶜᵖ{}
+		for _, l := range info.PpEnabledLayerNames().Slice(0, uint64(layerCount), l).MustRead(ctx, cmd, s, nil) {
+			if !isValidationLayer(l.String()) {
+				layers = append(layers, l)
+			}
+		}
+
+		validationLayersData := []api.AllocResult{}
+		for _, v := range validationLayers {
+			d := s.AllocDataOrPanic(ctx, v)
+			validationLayersData = append(validationLayersData, d)
+			layers = append(layers, NewCharᶜᵖ(d.Ptr()))
+		}
+		layersData := s.AllocDataOrPanic(ctx, layers)
+		info.SetEnabledLayerCount(uint32(len(layers)))
+		info.SetPpEnabledLayerNames(NewCharᶜᵖᶜᵖ(layersData.Ptr()))
+		infoData := s.AllocDataOrPanic(ctx, info)
```

Comment:
We need to clean this up somewhere.

Reply:
Fixed in 2b9a068.

Similar to read_framebuffer.go, the memory is cleaned after at the return of find_issue's Transform call. It is one-command later then freeing just after the call to mutate() for `vkCreateInstance`, because I inserted a `vkCreateDebugReportCallbackEXT` after it. But I think it should be fine, using defer looks nicer than calling free() manually after each mutate().